USING_ELEKTRA.md

```diff
@@ -0,0 +1,136 @@
+If you want to see all available configuration settings for the `server`, you can simply call
+```sh
+kdb ls '/sw/lcdproc/lcdd/#0/current/server'
+#> /sw/lcdproc/lcdd/#0/current/server/autorotate
+#> /sw/lcdproc/lcdd/#0/current/server/backlight
+#> /sw/lcdproc/lcdd/#0/current/server/bind
+#> /sw/lcdproc/lcdd/#0/current/server/driver
+...
+```
+To see all available configuration settings at all you can execute `kdb ls '/sw/lcdproc/lcdd/#0/current'`.
+To see the current value you can simply call `kdb get` on the respective key.
+Now lets take the curses driver for example again and lets see what is the 
+current background color is:
+```sh
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> cyan
+```
+The result comes from the set `default` metadata which is saved for the key. 
+You can see all metadata by calling `kdb lsmeta`:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> check/enum/#0
+#> check/enum/#1
+#> check/enum/#2
+#> check/enum/#3
+#> check/enum/#4
+#> check/enum/#5
+#> check/enum/#6
+#> check/enum/#7
+#> check/type
+#> default
+#> description
+#> type
+```
+Here we can als see `check/enum/#0-7` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum/#0'
+#> red
+```
+This `.../#<number>` notation is the typical array notation which is used by Elektra.
+If you want to query all metadata and see their values you can use this little script:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background' \
+| xargs -I% -n1 sh -c 'printf "% = " && kdb getmeta \
+"/sw/lcdproc/lcdd/#0/current/curses/background" "%"'
+#> check/enum/#0 = red
+#> check/enum/#1 = black
+#> check/enum/#2 = green
+#> check/enum/#3 = yellow
+#> check/enum/#4 = blue
+#> check/enum/#5 = magenta
+#> check/enum/#6 = cyan
+#> check/enum/#7 = white
+#> check/type = enum
+#> default = cyan
+#> description = background color when "backlight" is off
+#> type = enum
+```
+In future versions we might include a command which gives you all metadata 
+including their values with a simple command.
+If you try to set the value of the background to an invalid value, Elektra
+will raise an error and prevent a possible misconfiguration:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' purple
+#> Using name user/sw/lcdproc/lcdd/#0/current/curses/background
+#> The command kdb set failed while accessing the key database with the info:
+#> Sorry, the error (#121) occurred ;(
+#> Description: validation failed
+#> Reason: Validation of key "user/sw/lcdproc/lcdd/#0/current/curses/background" with string "purple" failed.
+#> Ingroup: plugin
+#> Module: enum
+#> At: <path>/src/plugins/enum/enum.c:218
+#> Mountpoint: user/sw/lcdproc/lcdd/#0/current
+#> Configfile: <path>/.config/LCDd.conf.17145:1549993505.561040.tmp
+```
```

Comment:
kdb editor? qt-gui? web-ui?...

Reply:
I added kdb editor and the qt-gui.

Concerning the web gui I would not feel very well by adding it as of now. It is still too error prone and would spoil users in regards of using elektra.

1. most difficult setup (node, yail, run-elektrad, ...). Very error prone and fix-heavy
2. Permanent error message #2408
3. Circumvents specifications and allows illegal values
4. Does not show all metadata which should be the actual use case for LCDproc